### PR TITLE
Eliminate n+1 queries in multiple problematic actions

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

File: app/models/version.rb [hunk 1]
```diff
@@ -4,7 +4,9 @@ class Version < ApplicationRecord # rubocop:disable Metrics/ClassLength
   RUBYGEMS_IMPORT_DATE = Date.parse("2009-07-25")
 
   belongs_to :rubygem, touch: true
-  has_many :dependencies, -> { order("rubygems.name ASC").includes(:rubygem) }, dependent: :destroy, inverse_of: "version"
+  has_many :dependencies, lambda {
+                            order(Rubygem.arel_table["name"].asc).includes(:rubygem).references(:rubygem)
+                          }, dependent: :destroy, inverse_of: "version"
```
Comment (indirect) [inline app/models/version.rb:9]: I don't understand what's different here... why switch from `->` to lambda, and why use `arel_table`? also, what does adding `references` do?

**Review by indirect — APPROVED** [review]
nice work! I'd love to see some graphs of how our requests times change after this.

File: app/models/version.rb (hunk 1 above)
Comment (segiddins) [reply app/models/version.rb:9]: no functional change here -- i think it got rewritten while I was trying out something else. the nice this is this avoids SQL strings by using the arel table.

Comment (simi) [reply app/models/version.rb:9]: @indirect `references` ensures data are loaded using join and not by second query. In this case `eager_load` could be used as well to make it shorter. :thinking:

**Pull request merged**